### PR TITLE
Update information about nix configuration

### LINKED ISSUE #3028: Research, check and document what Nix config is needed to get GHC from cache while building the project for the first time
As observed by @nfrisby in https://github.com/input-output-hk/ouroboros-network/pull/2885 cachix is most likely not needed to get right cache hits while building our project.
@EncodePanda also points out that `substitutors` section in `Onboarding.md` seems correct, but it is different than what haskell.nix documentation is proposing.

This ticket is done when:
1. proper configuration for the Nix build is found, such that: when having a clean nix store, upon building the project, GHC compiler is *downloaded* from the cache and *not* build from the scratch
2. this configuration is properly documented in `Onboarding.md`

### PR DESCRIPTION
**Pull request by EncodePanda**
Fixes #3028.

### CODE REVIEW

File: ouroboros-consensus/docs/Onboarding.md [hunk 1]
```diff
@@ -31,24 +31,18 @@ nix-env (Nix) 2.3.10
 
 ##### 2. cachix
```
Comment (nfrisby) [inline ouroboros-consensus/docs/Onboarding.md:32]: Rename this to "caching" or maybe "substituters"?

**Review by nfrisby — CHANGES_REQUESTED** [review]
Nice simplification! One last hiccup: the section title is stale now.

**Review by Jasagredo — APPROVED** [review]
Apart from @nfrisby comment, looks good to me.

**Comment by EncodePanda** [comment]
bors merge

**Comment by EncodePanda** [comment]
@nfrisby this is ready for review now

**Review by nfrisby — APPROVED** [review]
Nice 👍 

Approved

**Comment by EncodePanda** [comment]
bors merge

**Pull request merged**